keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -345,6 +346,9 @@ allocated, then the IP address chosen as the default IP address will match the
 IP family of the cluster's configured service CIDR. For example, if the service
 CIDR is IPv4, then the IPv4 address will be used as the default address.
 
+It's the CNI plugin responsibility to return the default IP address corresponding
+to the cluster's default IP family.
```

Comment:
I think the current mood is that there is no "cluster default IP family". Dual-stack-ness is local/distributed, not global/coordinated:

- the service CIDR can be single-stack IPv4, single-stack IPv6, dual-stack IPv4-primary, or dual-stack IPv6-primary
- the cluster CIDR can be single-stack IPv4, single-stack IPv6, dual-stack IPv4-primary, or dual-stack IPv6-primary
- each individual node can have node IPs that are single-stack IPv4, single-stack IPv6, dual-stack IPv4-primary, or dual-stack IPv6-primary
- load balancers...
- etc?

Not because we necessarily _want_ that, but because we have no higher-level coordination of configuration, so we can't _avoid_ that.

So this basically becomes "It's the CNI plugin's responsibility to return the IPs in the order that the CNI plugin has been configured to return the IPs in", which doesn't really need to be said...

Reply:
https://github.com/kubernetes/kubernetes/issues/95251